### PR TITLE
Add prettier as dev dep and pre-commit hook to run it

### LINKED ISSUE #127: Prettier config/set up
I noticed that there is a `.prettierrc` file in the frontend folder, but i don't see prettier as a dev dependency. Is it actually set up to run at all? if not, can I make a PR to
1. add it as a dev dep
2. run it on all the frontend files
3. potentially add a pre-commit hook to format staged files to catch if any contributors make changes but do not run prettier

lmk what you think

### PR DESCRIPTION
**Pull request by orteth01**
fixes #127 

I went with option 1 in the [prettier pre-commit hook docs](https://prettier.io/docs/en/precommit.html#option-1-lint-stagedhttpsgithubcomokonetlint-staged)

### CODE REVIEW

File: frontend/ui/lobby.tsx [hunk 1]
```diff
@@ -24,43 +26,41 @@ export const Lobby = ({ defaultGameID }) => {
...
+        window.location = newURL;
+      });
   }
 
   let toggleWordSet = (wordSet) => {
-    let wordSets = [ ...selectedWordSets ];
+    let wordSets = [...selectedWordSets];
     let index = wordSets.indexOf(wordSet);
 
-    if index == -1 {
```
Comment (orteth01) [inline frontend/ui/lobby.tsx:63]: this was missing parens

File: frontend/ui/lobby.tsx [hunk 2]
```diff
@@ -108,30 +112,37 @@ export const Lobby = ({ defaultGameID }) => {
...
                 words={customWordsText}
-                onWordChange = {(w) => {
+                onWordChange={(w) => {
                   setCustomWordsText(w);
-                  setWords({...words, 'Custom': (w
-                    .trim()
-                    .split(',')
-                    .map(w => w.trim())
-                    .filter(w => w.length > 0))});
-                }
```
Comment (orteth01) [inline frontend/ui/lobby.tsx:132]: there was a missing closing bracket here

**Review by orteth01 — COMMENTED** [review]
Just want to highlight two actual code changes i made (prettier was breaking due to some syntax errors). everything else is either prettier formatting or devDep/pre-commit configuration

**Comment by jbowens** [comment]
lgtm, thanks!

**Pull request merged**